packages/start/src/config/index.ts

```diff
@@ -521,11 +600,164 @@ function tsrRoutesManifest(opts: {
           routes,
         }
 
-        if (process.env.TSR_VITE_DEBUG)
+        if (process.env.TSR_VITE_DEBUG) {
           console.info(JSON.stringify(routesManifest, null, 2))
+        }
 
         return `export default () => (${JSON.stringify(routesManifest)})`
       }
+      if (id === 'tsr:api-manifest') {
+        return `export default () => ({
+            apiRoutes: {}
+          })`
+      }
+      return
+    },
+  }
+}
+
+function tsrAPIRouter(opts: {
+  tsrConfig: z.infer<typeof configSchema>
+  apiBase: string
+}) {
+  const checkExp = new RegExp(`${getAPIBaseSegment(opts.apiBase)}/`)
+  return function (router: VinxiRouterSchemaInput, app: VinxiAppOptions) {
+    // Our own custom router that extends the VinxiBaseFileSystemRouter
+    // to full necessary API routes into its own "bundle"
+    class StartAPIFsRouter extends VinxiBaseFileSystemRouter {
+      toPath(src: string): string {
+        const inputPath = vinxiFsRouterCleanPath(src, this.config)
+
+        const segments = startAPIRouteSegmentsFromPath(inputPath)
+
+        const pathname = segments
+          .map((part) => {
+            if (part.type === 'splat') {
+              return `*splat`
+            }
+
+            if (part.type === 'param') {
+              return `:${part.value}?`
+            }
+
+            return part.value
+          })
+          .join('/')
+
+        return pathname.length > 0 ? `/${pathname}` : '/'
+      }
+
+      toRoute(src: string) {
+        const webPath = this.toPath(src)
+
+        if (!checkExp.test(webPath)) {
+          return undefined
+        }
+
+        const [_, exports] = vinxiFsRouterAnalyzeModule(src)
+
+        const hasRoute = exports.find((exp) => exp.n === 'Route')
+
+        return {
+          path: webPath,
+          filePath: src,
+          $route: hasRoute
+            ? {
+                src,
+                pick: ['Route'],
+              }
+            : undefined,
+        }
+      }
+    }
+
+    return new StartAPIFsRouter(
+      {
+        dir: opts.tsrConfig.routesDirectory,
+        extensions: ['js', 'jsx', 'ts', 'tsx'],
+      },
+      router,
+      app,
+    )
+  }
+}
+
+function tsrAPIManifest(opts: {
+  tsrConfig: z.infer<typeof configSchema>
+  apiBase: string
+}): vite.Plugin {
+  let config: vite.ResolvedConfig
+
+  return {
+    name: 'tsr-api-manifest',
+    configResolved(resolvedConfig) {
+      config = resolvedConfig
+    },
+    resolveId(id) {
+      if (id === 'tsr:routes-manifest') {
+        return id
+      }
+      if (id === 'tsr:api-manifest') {
+        return id
+      }
+      return
+    },
+    load(id) {
+      if (id === 'tsr:api-manifest') {
+        // If we're in development, return a dummy manifest
+        // if (config.command === 'serve') {
+        //   return `export default () => ({
+        //     apiBase: '${opts.tsrConfig.apiBase}',
+        //     apiRoutes: {}
+        //   })`
+        // }
+
```

Comment:
@tannerlinsley I think we need to figure out how to get this built `manifest.json` over to the `api-manifest`. Or in the `handleAPIFileRoute` function, we need to get the vinxi API manifest in there.

This is what vinxi's manifest for API looks like (located in `.vinxi/build/api/api/.vite/manifest.json`).

```json
{
  "_createApiHandler-BWUdrWLi.js": {
    "file": "assets/createApiHandler-BWUdrWLi.js",
    "name": "createApiHandler"
  },
  "_json-BfrH-og-.js": {
    "file": "assets/json-BfrH-og-.js",
    "name": "json"
  },
  "app/routes/api.boo.$id.name.$.tsx?pick=Route": {
    "file": "api.boo._id.name._.js",
    "name": "api.boo._id.name._",
    "src": "app/routes/api.boo.$id.name.$.tsx?pick=Route",
    "isEntry": true,
    "imports": [
      "_json-BfrH-og-.js",
      "_createApiHandler-BWUdrWLi.js"
    ]
  },
  "app/routes/api/foo.$.ts?pick=Route": {
    "file": "foo._.js",
    "name": "foo._",
    "src": "app/routes/api/foo.$.ts?pick=Route",
    "isEntry": true,
    "imports": [
      "_json-BfrH-og-.js",
      "_createApiHandler-BWUdrWLi.js"
    ]
  },
  "app/routes/api/hello.ts?pick=Route": {
    "file": "hello.js",
    "name": "hello",
    "src": "app/routes/api/hello.ts?pick=Route",
    "isEntry": true,
    "imports": [
      "_json-BfrH-og-.js",
      "_createApiHandler-BWUdrWLi.js"
    ]
  },
  "virtual:$vinxi/handler/api": {
    "file": "api.js",
    "name": "api",
    "src": "virtual:$vinxi/handler/api",
    "isEntry": true,
    "imports": [
      "_createApiHandler-BWUdrWLi.js"
    ]
  }
}
```

Once we get that, in the `handleAPIFileRoute` function, it'll just be a matter of handling the import (similar what's already there for server-functions).